docs/migrating/from-gatsby.md

```diff
@@ -0,0 +1,235 @@
+---
+description: Learn how to transition an existing Gatsby project to Next.js.
+---
+
+# Migrating from Gatsby
+
+This guide will help you understand how to transition from an existing Gatsby project to Next.js. Since both frameworks are built on top of React, the transition can be broken down into a series of steps. Migrating to Next.js will allow you to:
+
+- Choose which [data fetching](/docs/basic-features/data-fetching) strategy you want on a per-page basis (instead of only SSG)
+- Use [Incremental Static Regeneration](/docs/basic-features/data-fetching#incremental-static-regeneration) to update _existing_ pages by re-rendering them in the background as traffic comes in
+- Use [API Routes](/docs/api-routes/introduction)
+
+And more! First, let's talk about running a Next.js application.
+
+## Local Development
+
+Gatsby applications start using `gatsby develop` and run at `localhost:8000`. To create and start a production build, you run `gatsby build && gatsby serve`. Your compiled code is located at `public/`.
+
+Next applications start using `next dev` and run at `localhost:3000`. To create a start a production build, run `next build && next start`. Your compiled code is located at `.next/`.
+
+The first step towards migrating to Next.js is to uninstall all related Gatsby packages and install `next`. Do not remove `react` or `react-dom` from your `package.json`. Here's an example `package.json`:
```

Comment:
I think we’d want to reorganize the above paragraphs a bit. What do you think about something like this?

---

The first step towards migrating to Next.js is to update `package.json` and dependencies. You should:

- Remove all Gatsby-related packages (but keep `react` and `react-dom`).
- Install `next`.
- Add Next.js related commands to `scripts`. One is `next dev`, which runs a development server at `localhost:3000`. You should also add `next build` and `next start` for creating and starting a production build.

Here's an example `package.json`:

Reply:
I think it’d also be helpful if you could link to the `package.json` diff in your PR:

https://github.com/leerob/gatsby-to-nextjs/pull/1/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2